packages/nvidia_nat_core/src/nat/data_models/atif/trajectory.py

```
@@ -0,0 +1,132 @@
+            if step.observation is None:
+                continue
+            tool_call_ids = set()
+            if step.tool_calls:
+                tool_call_ids = {tc.tool_call_id for tc in step.tool_calls}
+            for result in step.observation.results:
+                if result.source_call_id is not None and result.source_call_id not in tool_call_ids:
+                    raise ValueError(
+                        f"Observation result references source_call_id "
+                        f"'{result.source_call_id}' which is not found in "
+                        f"step {step.step_id}'s tool_calls"
+                    )
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)